--- a/packages/opencode/src/cli/cmd/tui/thread.ts	2025-11-03 17:00:01.752000789 +1100
+++ b/packages/opencode/src/cli/cmd/tui/thread.ts	2025-11-03 17:00:01.775001147 +1100
@@ -2,6 +2,7 @@
 import { tui } from "./app"
 import { Rpc } from "@/util/rpc"
 import { type rpc } from "./worker"
+import { Server } from "../../../server/server"
 import { upgrade } from "@/cli/upgrade"
 import { Session } from "@/session"
 import { bootstrap } from "@/cli/bootstrap"
@@ -82,19 +83,11 @@
         return undefined
       })()
 
-      const worker = new Worker("./src/cli/cmd/tui/worker.ts")
-      worker.onerror = console.error
-      const client = Rpc.client<typeof rpc>(worker)
-      process.on("uncaughtException", (e) => {
-        console.error(e)
-      })
-      process.on("unhandledRejection", (e) => {
-        console.error(e)
-      })
-      const server = await client.call("server", {
-        port: args.port,
-        hostname: args.hostname,
+      const server = Server.listen({
+        port: args.port ?? 0,
+        hostname: args.hostname ?? "127.0.0.1",
       })
+
       await tui({
         url: server.url,
         sessionID,
@@ -102,7 +95,8 @@
         agent: args.agent,
         prompt: args.prompt,
         onExit: async () => {
-          await client.call("shutdown", undefined)
+          await import("@/global").then((mod) => mod.Global.disposeAll())
+          server.stop()
         },
       })
     })
